compiler/src/org.graalvm.compiler.replacements.test/src/org/graalvm/compiler/replacements/test/StringCompressInflateTest.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (c) 2018, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.graalvm.compiler.replacements.test;
+
+import org.graalvm.compiler.nodes.StructuredGraph;
+import org.graalvm.compiler.replacements.StringSubstitutions;
+import org.graalvm.compiler.replacements.amd64.AMD64StringLatin1InflateNode;
+import org.graalvm.compiler.replacements.amd64.AMD64StringUTF16CompressNode;
+import org.graalvm.compiler.test.AddExports;
+import org.junit.Test;
+
+import java.io.UnsupportedEncodingException;
+
+import jdk.vm.ci.amd64.AMD64;
+import jdk.vm.ci.code.InstalledCode;
+import jdk.vm.ci.meta.ResolvedJavaMethod;
+
+
+/**
+ * Test intrinsic/node substitutions for (innate) methods StringLatin1.inflate
+ * and StringUTF16.compress provided by {@link AMD64StringLatin1Substitutions}
+ * and {@link AMD64StringUTF16Substitutions}.
+ */
+@AddExports({"java.base/java.lang"})
+public final class StringCompressInflateTest extends MethodSubstitutionTest {
+
+    final static int N = 1000;
+
+    @Test
+    public void testStringLatin1Inflate() throws ClassNotFoundException,
+                                                 UnsupportedEncodingException
+    {
+        if (Java8OrEarlier)
```

Comment:
Consider using `org.junit.Assume.assumeFalse(Java8OrEarlier)` instead for this kind of thing.

Reply:
I found the use in the code to be inconclusive, thus adopting the one more clear to me, with obvious operational interpretation (as opposed to the quasi-declarative version in *junit* using an implicit operational realisation, hiding the control-flow -- which I consider a rather poor approach). I don't think the "junit.Assume" support adds anything in this case.

Reply:
The main difference is that the junit output includes a message indicating the test passed because of an assumption.  assumeTrue is broadly how we disable tests which only run on certain configurations.

Reply:
Fair point.